Tests/SwiftSyntaxMacroExpansionTest/MemberMacroTests.swift

```diff
@@ -299,4 +299,29 @@ final class MemberMacroTests: XCTestCase {
       ]
     )
   }
+  
+  func testMemberMacroOnVar() {
+    struct TestMacro: MemberMacro {
+      static func expansion(
+        of node: AttributeSyntax,
+        providingMembersOf declaration: some DeclGroupSyntax,
+        conformingTo protocols: [TypeSyntax],
+        in context: some MacroExpansionContext
+      ) throws -> [DeclSyntax] {
+        return []
+      }
+    }
+
+    assertMacroExpansion(
+      """
+      @Test var x: Int
+      """,
+      expandedSource: """
+        var x: Int
+        """,
+      macros: [
+        "Test": TestMacro.self
+      ]
+    )
```

Comment:
I think you have it the wrong way round. From the issue description

> the following test case should emit an error. 

Also: Did you run all the other tests? I suspect all member tests will fail with your current changes.

Reply:
> I think you have it the wrong way round. From the issue description

You mean the test itself is wrong or it's in the wrong place?

> Also: Did you run all the other tests? I suspect all member tests will fail with your current changes.

Well yes, as I said other member tests are failing.

Reply:
> You mean the test itself is wrong or it's in the wrong place?

Yes, having an attached member macro on a variable should emit an error.

Reply:
Ok so I had mistakenly put a `guard` instead of `if`, although I am yet to find the `trigger` for `VariableDeclSyntax` so I can see the test failing and then add the error message, the previous errors and tests are good which weren't supposed to fail.

Reply:
What `diagnostic` do you recommend adding here? Is there an existing one for such case or we need a custom addition to the `enum` which we then `throw` during detection?

Reply:
What property should be check under `MacroExpansion` in the below case (directed here from `MacroSystem` -> `expandAttachedMacro` -> `expandAttachedMacroWithoutCollapsing`) if I am under the right place:

```Swift
case (let attachedMacro as MemberMacro.Type, .member):
// ...
```

Reply:
In the issue I wrote 
> specifically, if the declaration is not a `DeclGroupSyntax`

So that’s the syntax node type I would check.

Reply:
This check is for the `declGroup` I suppose, not for the actual check if it is a `variable`?

Also, could you briefly differentiate between `attributeNode`, `declarationNode` and `declGroup`?

Reply:
@ahoppen 
According to the issue, as you mentioned:

> specifically, if the declaration is not a DeclGroupSyntax

But, the following check exists:

```Swift
guard let declGroup = declarationNode.asProtocol(DeclGroupSyntax.self)
```

How does it not throw an error here if the node isn't a `DeclGroupSyntax`?